contrib/mesos/pkg/scheduler/scheduler.go

```diff
@@ -449,7 +449,7 @@ func (k *KubernetesScheduler) StatusUpdate(driver bindings.SchedulerDriver, task
 func (k *KubernetesScheduler) reconcileTerminalTask(driver bindings.SchedulerDriver, taskStatus *mesos.TaskStatus) {
 	task, state := k.taskRegistry.UpdateStatus(taskStatus)
 
-	if (state == podtask.StateRunning || state == podtask.StatePending) && taskStatus.SlaveId != nil &&
+	if (state == podtask.StateRunning || state == podtask.StatePending) &&
```

Comment:
I was originally thinking that we should handle TASK_ERROR like we do for TASK_LOST, TASK_KILLED; not TASK_FAILED. TASK_ERROR is pretty serious: it means that the apiserver has accepted a pod for scheduling that we can probably *never* schedule.

for a one-off pod, this is less problematic; we can log an error, delete the pod, and fire a k8s "event" indicating what happened.

for a pod that's being launched by a replication controller  .. things seem a little stickier. if we kill the pod, the rc will just create another one. i don't know if the RC has any backoff logic for re-creating pods that constantly fail. if we generate an API server event for each failure .. that could be pretty noisy.

perhaps instead of deleting the pod we could update it's pod/status to Failed and leave a helpful message in there. maybe an RC would respond better to consecutive pod failures?